### PR TITLE
init commit

### PR DESCRIPTION
**Pull request by entertainment-veks**

### CODE REVIEW

File: src/main/java/hibernate/service/AuthenticationService.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+public interface AuthenticationService {
+    User login(String email, String password) throws AuthenticationException;
+
+    /**
+     * We should register a new user. New user entity will contains the email and password
+     * @param email - user email. should be unique for each user
+     * @param password - user password
+     * @return new user instance
+     */
```
Comment (Fedich09) [inline src/main/java/hibernate/service/AuthenticationService.java:14]: I think we should delete this
```suggestion
```

File: src/main/java/hibernate/service/impl/AuthentificationServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+@Service
+public class AuthentificationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> dirty = userService.findByEmail(email);
+        if (dirty.isEmpty()) {
+            throw new AuthenticationException("User doesn't exist");
```
Comment (Fedich09) [inline src/main/java/hibernate/service/impl/AuthentificationServiceImpl.java:21]:
```suggestion
            throw new AuthenticationException("User doesn't exist");
```
try to use just one if block and one exception

**Review by Fedich09 — CHANGES_REQUESTED** [review]

**Review by Fedich09 — APPROVED** [review]

File: src/main/java/hibernate/service/impl/AuthentificationServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> dirty = userService.findByEmail(email);
+        if (dirty.isPresent() && dirty.get().getPassword()
+                .equals(SecurityUtil.hashPassword(password, dirty.get().getSalt()))) {
+            return dirty.get();
+        }
+        throw new AuthenticationException("User doesn't exist");
```
Comment (dmytro-hryhoruk) [inline src/main/java/hibernate/service/impl/AuthentificationServiceImpl.java:24]:
```suggestion
        throw new AuthenticationException("Wrong email or password");
```

Comment (dmytro-hryhoruk) [reply src/main/java/hibernate/service/impl/AuthentificationServiceImpl.java:24]: user may exist but the password just wrong

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**